tests/foreman/longrun/test_inc_updates.py

```diff
@@ -0,0 +1,310 @@
+                u'content_label': PRD_SETS['rhel6_sat6tools']['label'],
+                u'value': u'1'
+            }})
+
+        # Step 16: Create three client machines and register them to satellite
+        cls.vm = []
+        for _ in range(3):
+            cls.vm.append(VirtualMachine(distro='rhel67', tag='incupdate'))
+
+        cls.setup_vm(cls.vm[0], rhel_6_ak.name, cls.org.label)
+        for i in range(3):
+            cls.setup_vm(cls.vm[i], rhel_6_partial_ak.name, cls.org.label)
+
+        # Find the content hosts (systems) and save them
+        systems = System(organization=cls.org).search()
+        cls.systems = []
+        cls.partial_systems = []
+
+        for system in systems:
+            if system.content_view.read().name == cls.rhel_6_cv.name:
+                cls.systems.append(system)
+            else:
+                cls.partial_systems.append(system)
+
+    @classmethod
+    def tearDownClass(cls):
+        """Destroys all provisioned vms"""
+        for virt_machine in cls.vm:
+            virt_machine.destroy()
+        entity_mixins.TASK_TIMEOUT = cls.old_task_timeout
+
+    @staticmethod
+    def setup_vm(client, act_key, org_name):
+        """Creates the vm and registers it to the satellite"""
+        client.create()
+        client.install_katello_cert()
+
+        # Register content host, install katello-agent
+        result = client.register_contenthost(
+            act_key,
+            org_name,
+            releasever='6.7'
+        )
+        assert result.return_code == 0
+        result = client.install_katello_agent()
+        client.run('katello-package-upload')
+        return
+
+    @staticmethod
+    def get_applicable_errata(repo):
+        """Retrieves applicable errata for the given repo"""
+        return Errata(repository=repo).search(
+            query={'errata_restrict_applicable': True}
+        )
+
+    def test_api_inc_update_noapply(self):
+        """@Test: Check if api incremental update can be done without
+        actually applying it
+
+        @Feature: Incremental Update
+
+        @Setup:  The prerequisites are already covered in the setUpClass() but
+        for easy debug, get the content view id, Repository id and Lifecycle
+        environment id using hammer and plug these statements on the top of the
+        test. For example::
+
+            self.rhel_6_partial_cv = ContentView(id=38).read()
+            self.rhel_66_repo = Repository(id=164).read()
+            self.qe_lce = LifecycleEnvironment(id=46).read()
+
+        @Assert: Incremental update completed with no errors and Content view
+        has a newer version
+        """
+
+        # Get the content view versions and find the recent one.  API always
+        # returns the versions in ascending order so it is safe to assume the
+        # last one in the list is the recent
+        cv_versions = self.rhel_6_partial_cv.version
+        cvv_id = cv_versions[len(cv_versions) - 1].id
```

Comment:
This can be changed to:

```python
cvv_id = cv_versions[-1].id
```

That said, I think the `cvv_id` variable can be dropped entirely. Below, it's pretty sane to write:

```python
        # Apply incremental update using the first applicable errata
        ContentViewVersion().incremental_update(data={
            'content_view_version_environments': [{
                'content_view_version_id': cv_versions[-1].id,
                'environment_ids': [self.qe_lce.id]
            }],
            'add_content': {
                'errata_ids': [errata_list[0].id]
            }
        })
```